connman/vpn/plugins/wireguard.c

```diff
@@ -338,6 +409,35 @@ static gboolean wg_dns_reresolve_cb(gpointer user_data)
 			info->device.name);
 
 	run_dns_reresolve(info);
+}
+
+static gboolean wg_dns_reresolve_cb(gpointer user_data)
+{
+	struct wireguard_info *info = user_data;
+
+	DBG("");
+
+	if (info->resolv_id > 0) {
+		DBG("previous query was running, abort it");
+		remove_resolv(info);
+	}
+
+	info->resolv = vpn_util_resolve_new(0);
+	if (!info->resolv) {
+		connman_error("cannot create GResolv");
+		return G_SOURCE_REMOVE;
+	}
+
+	info->resolv_id = vpn_util_resolve_hostname(info->resolv,
+						info->endpoint_fqdn,
+						resolve_endpoint_cb, info);
+
+	if (info->resolv_id <= 0) {
```

Comment:
Hm, this is halfway useless check since it is a `guint`.

Reply:
also that special return from here just before doing the same return anyway seems a bit useless

Reply:
https://github.com/sailfishos/connman/blob/master/connman/gweb/gresolv.c#L1030 well, interesting, return is `guint` but it does return `-EIO` as well. Classic. I guess this needs to be addressed as well.

Reply:
Oh well, a workaround to get the error and return 0 whenever an error happens: b857c8f6bdec18c090d3e15ab9e02038fdf674c4 - it now makes the VPN die as those errors are quite severe (EIO).